lib/Backend/PreLowerPeeps.cpp

```diff
@@ -13,6 +13,13 @@ IR::Instr *Lowerer::PreLowerPeepInstr(IR::Instr *instr, IR::Instr **pInstrPrev)
 
     switch (instr->m_opcode)
     {
+#if defined(_M_IX86) || defined(_M_X64)
+        // this sort of addressing mode magic only applies to x86 platforms
```

Comment:
>// this sort of addressing mode magic only applies to x86 platforms [](start = 8, length = 67)

ARM has some similar instruction form as `ADD r4, r4, r4, LSL #4`, could we also peep that?